File: demo_core_av/app/src/main/java/com/bandyer/demo_core_av/StatsPagerAdapter.java
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_core_av;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.HashMap;
import java.util.LinkedHashMap;

/**
 * A simple pager adapter that represents Stats pages.
 */
public class StatsPagerAdapter extends PagerAdapter {


    private HashMap<String, StatsPage> mPages = new LinkedHashMap<>();
    private HashMap<Integer, String> mPagesIdentifiers = new LinkedHashMap<>();

    private Context mContext;

    public StatsPagerAdapter(Context context) {
        mContext = context;
    }

    public StatsPage addOrGetPage(String id) {
        if (mPages.containsKey(id)) return mPages.get(id);
        mPagesIdentifiers.put(mPagesIdentifiers.size(), id);
        StatsPage page = new StatsPage();
        mPages.put(id, page);
        notifyDataSetChanged();
        return page;
    }

    @NonNull
    @Override
    @SuppressWarnings("ConstantConditions")
    public Object instantiateItem(@NonNull ViewGroup collection, int position) {
        return mPages.get(mPagesIdentifiers.get(position)).onCreateView(LayoutInflater.from(mContext), collection);
    }

    @Override
    public void destroyItem(@NonNull ViewGroup collection, int position, @NonNull Object view) {
        collection.removeView((View) view);
    }

    @Override
    public int getCount() {
        return mPages.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mPagesIdentifiers.get(position);
    }
}

File: demo_core_av/app/src/main/java/com/bandyer/demo_core_av/StatsPagerAdapter.kt
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */
package com.bandyer.demo_core_av

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewpager.widget.PagerAdapter
import java.util.*

/**
 * A simple pager adapter that represents Stats pages.
 */
class StatsPagerAdapter(private val mContext: Context) : PagerAdapter() {
    private val mPages: HashMap<String, StatsPage?> = LinkedHashMap()
    private val mPagesIdentifiers: HashMap<Int, String> = LinkedHashMap()
    fun addOrGetPage(id: String): StatsPage? {
        if (mPages.containsKey(id)) return mPages[id]
        mPagesIdentifiers[mPagesIdentifiers.size] = id
        val page = StatsPage()
        mPages[id] = page
        notifyDataSetChanged()
        return page
    }

    override fun instantiateItem(collection: ViewGroup, position: Int): Any {
        return mPages[mPagesIdentifiers[position]]!!.onCreateView(LayoutInflater.from(mContext), collection)
    }

    override fun destroyItem(collection: ViewGroup, position: Int, view: Any) {
        collection.removeView(view as View)
    }

    override fun getCount(): Int {
        return mPages.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mPagesIdentifiers[position]
    }

}